src/python/verifier/gates.py

```diff
@@ -19,6 +19,51 @@ def neg(a):
     return cos_a, -sin_a
 
 
+def mult(x, y):
+    # In this special case, both the lhs and the rhs are numbers.
+    if isinstance(x, (int, float)) and isinstance(y, (int, float)):
+        return x * y
+
+    # To apply trigonmetric angle formulas, one side must be a number.
+    # Without loss of generality, the left-hand side will be a number.
+    assert isinstance(x, (int, float)) or isinstance(y, (int, float))
+    if isinstance(y, (int, float)):
+        x, y = y, x
+
+    # The only angle formulas expressible in NLRA are integer multiples of angles.
```

Comment:
Whoops: non-linear real arithmetic.

Reply:
I see. I prefer not using the acronym here :)

Reply:
Btw this seems not accurate: there is also half-angle formula:
sin(x/2) = ±sqrt((1-cos(x))/2)
cos(x/2) = ±sqrt((1+cos(x))/2)
The signs can be derived from the signs of sin(x) and cos(x).

Reply:
Oh, cool. I didn't realize there was a `z3.Sqrt` function. Though I imagine the constraints would be really hard to solve when the parameters are symbolic.

In any case, I'll update the commit. Thanks!

Reply:
@xumingkuan How were you thinking of determining `sgn(cos(x/2))` from `cos(x)` and `sin(x)`.

At first, I was thinking that `sgn(sin(x/2)) = 1` and `sgn(cos(x/2)) = z3.If(sin(x) >= 0, 1, -1)`. Then I realized that this is only true when `x` is in `[0, 2pi]`. For example, if `x = pi/2 + 2pi`, then `x/2 = pi/4 + pi`, and as a result `sgn(sin(x)) = 1 =\= -1 = sgn(cos(x/2))`.

Can we assume the verifier only works with angles from `[0, 2pi]`?

Reply:
Good question! I would also assume that the verifier only works with angles in `[0, 2pi)` but there are cases that the user needs to be aware of. For example, when computing `1.5 * x`, `(3 * x) / 2` and `x + (x / 2)` may have different results.

Unless you need non-integer multiples, I prefer only supporting multiplying integer values, with a comment that dividing by 2 is feasible but the sign needs to be determined. It might be better to have another `HalfGate` (with only one input `x` and one output `x / 2`) for the user to specify if they want `(3 * x) / 2` or `x + (x / 2)` in the future.

Reply:
Yeah, I think a `HalfGate` would make more sense.

Personally, I don't have a use-case for the `HalfGate`. I'm happy to implement it and open a new PR if you think it would be useful though.

Reply:
Actually, I don't think `HalfGate` will be useful -- it seems to me that it's just better to have the symbolic parameter `y = x / 2` and use `2 * y` whenever the user needs `x`.